### PR TITLE
Returns the longest word in that sentence

### PR DESCRIPTION
**Pull request by ozgurozbekuk**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [ ] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [x] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [x] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [x] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [ ] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
  - _It's ok to have extra helper functions if necessary, like with mergesort_
- [x] The function has correct type annotations
- [ ] The function is not called at the top level of the function file 
  - _Recursive solutions **can** call the function from **inside** the function body_

### Do's

- [x] Variable names help to understand the strategy
- [x] Any comments are clear and describe the strategy
- [x] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in any docstrings or tests
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [x] Variables are named with snake_case
- [x] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] There are no commented lines of code
- [x] There are no `print` statements anywhere
- [x] The code includes defensive assertions
- [x] Defensive assertions include as little logic as possible

### CODE REVIEW

**Comment by Mo-Altayeb** [comment]
Don't forget to link the PR to the issue and assign yourself to it. But I got you covered this time 😄

File: solutions/longest_word.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+"""
```
Comment (Mo-Altayeb) [inline solutions/longest_word.py:1]: Add a module header before the module docstring
it looks like this
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

```

File: solutions/tests/test_longest_word.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
+"""
```
Comment (Mo-Altayeb) [inline solutions/tests/test_longest_word.py:1]: Same thing here
Add
```
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
```

File: solutions/longest_word.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+
+    Args:
+        sentence (str): A sentence from which to find the longest word.
+
+    Returns:
+        str: The longest word in the sentence. If there are multiple longest words,
+             the first one encountered is returned.
+
+    Raises:
+        ValueError: If the input is not a string or if the string is empty.
```
Comment (Mo-Altayeb) [inline solutions/longest_word.py:29]: I think it raises AssertionError and not ValueError

File: solutions/longest_word.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    Example:
+        >>> longest_word("I love programming with Python")
+        'programming'
+
+        >>> longest_word("apple banana cherry!")
+        'banana'
+
+        >>> longest_word("Hello, world!")
+        'world'
```
Comment (Mo-Altayeb) [inline solutions/longest_word.py:43]: One minor error here. the expected value for this doctest should be 'hello' and not 'world'.
Sorry for not noticing this and commenting about it yesterday.

**Review by Mo-Altayeb — COMMENTED** [review]
All looks good now

**Comment by Mo-Altayeb** [comment]
Actually the change I meant is in the ``` solutions/longest_word.py ```  file. Not the unit tests file. Go to line 43 in that file and change
'world' 
to
'Hello'
and that should do it 😄

**Review by Mo-Altayeb — APPROVED** [review]
Perfect! 😃
